web_external/js/views/body/MapPanel.js

```diff
@@ -115,6 +134,10 @@ minerva.views.MapPanel = minerva.View.extend({
             this.legendWidget[datasetId].remove(datasetId);
             delete this.legendWidget[datasetId];
         }
+        var layerIndex = $.inArray(layer, this.map.featureInfoWidget.layers);
```

Comment:
I moved this block of code inside the following conditional block that makes sure the layer being removed is actually a WMS layer, and also added a check to make sure the featureInfoWidget exists.

I can create the widget inside the MapPanel's init method, but my thinking was that it should be created only if a WMS layer is added to the map.  I'm happy to go with whichever option you prefer.